### PR TITLE
ans starter support Eureka/Consul register and add sms module

### PR DESCRIPTION
**Pull request by pbting**
1. ans starter support Eureka/Consul register
2. add sms module starter

### CODE REVIEW

File: spring-cloud-alibaba-examples/sms-example/pom.xml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+    <parent>
+        <groupId>org.springframework.cloud</groupId>
+        <artifactId>spring-cloud-alibaba-examples</artifactId>
+        <version>0.1.2.BUILD-SNAPSHOT</version>
+    </parent>
+
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>alibaba.com</groupId>
```
Comment (flystar32) [inline spring-cloud-alibaba-examples/sms-example/pom.xml]: no need to add groupId here, refer to oss-example

File: spring-cloud-alicloud-ans/src/main/java/org/springframework/cloud/alicloud/ans/endpoint/AnsEndpointAutoConfiguration.java [hunk 2]
```diff
@@ -25,7 +26,7 @@
  * @author xiaolongzuo
  */
 @ConditionalOnWebApplication
-@ConditionalOnClass(name = "org.springframework.boot.actuate.endpoint.AbstractEndpoint")
+@ConditionalOnClass(AbstractEndpoint.class)
```
Comment (flystar32) [inline spring-cloud-alicloud-ans/src/main/java/org/springframework/cloud/alicloud/ans/endpoint/AnsEndpointAutoConfiguration.java]: why use `AbstractEndpoint.class` here ,but use `@ConditionalOnClass(name = "org.springframework.boot.actuate.endpoint.AbstractEndpoint")` in MigrateEndpointAutoConfiguration

File: spring-cloud-alicloud-ans/src/main/java/org/springframework/cloud/alicloud/ans/migrate/MigrateRefreshEventListener.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+/**
+ * @author pbting
+ */
+@Component
+public class MigrateRefreshEventListener implements ApplicationListener<RefreshEvent> {
+	private final static Log log = LogFactory.getLog(MigrateRefreshEventListener.class);
+
+	private final static int CHECK_INTERVAL = 1;
+
+	private final static String MIGRATE_SWITCH = "sca.migrate.ans.switch";
```
Comment (flystar32) [inline spring-cloud-alicloud-ans/src/main/java/org/springframework/cloud/alicloud/ans/migrate/MigrateRefreshEventListener.java]: just a suggestion for use full name like `spring.cloud.alicloud.migarte.ans.switch`

Comment (pbting) [reply spring-cloud-alicloud-ans/src/main/java/org/springframework/cloud/alicloud/ans/migrate/MigrateRefreshEventListener.java]: ok,good suggestion

File: spring-cloud-alicloud-ans/src/main/java/org/springframework/cloud/alicloud/ans/endpoint/AnsEndpointAutoConfiguration.java (hunk 2 above)
Comment (pbting) [reply spring-cloud-alicloud-ans/src/main/java/org/springframework/cloud/alicloud/ans/endpoint/AnsEndpointAutoConfiguration.java]: goog ,use AbstractEndpoint.class or Endpoint.class

File: spring-cloud-alibaba-examples/sms-example/pom.xml (hunk 1 above)
Comment (pbting) [reply spring-cloud-alibaba-examples/sms-example/pom.xml]: has removed

**Pull request merged**